src/daft-physical-plan/src/physical_planner/translate.rs

```diff
@@ -1094,6 +1098,18 @@ pub fn populate_aggregation_stages(
                     ));
                 final_exprs.push(col(concat_of_list_id.clone()).alias(output_name));
             }
+            AggExpr::Set(e, ignore_nulls) => {
+                let list_agg_id =
+                    add_to_stage(AggExpr::List, e.clone(), schema, &mut first_stage_aggs);
+                let list_concat_id = add_to_stage(
+                    AggExpr::Concat,
+                    col(list_agg_id.clone()),
+                    schema,
+                    &mut second_stage_aggs,
+                );
+                let result = unique(col(list_concat_id.clone()), *ignore_nulls).alias(output_name);
+                final_exprs.push(result);
```

Comment:
Come to think of it, since the translation here converts it into a two-stage aggregation that uses list and then concat+unique, are we ever actually using `Series::agg_set` during an execution of agg_set? If we don't intend to, there's actually no need to implement that at all.

Reply:
It is pretty confusing how the expression that is called is not always the one that we actually execute in an agg. Let me know if you need me to explain any of this.